### PR TITLE
Introduce Floating Links in the Business Directory Panel

### PR DESCRIPTION
**Pull request by shervElmi**
This PR introduces the Floating Links feature to the Business Directory Panel, which provides users with a set of links for quick navigation to the **Upgrade**, **Support**, and **Documentation** pages.

## Testing Instructions:
1. Navigate to `WP Admin > Directory`.
2. Check the perfect pixel implementation.
  - **Figma File:** https://www.figma.com/file/YcSWkrjVGroC1zyxhuHH6W/Floating-Links?node-id=1-117&t=Xf5EKO8Mvs1JwuFj-4
3. Verify that all functionality works well.

### CODE REVIEW

**Comment by garretlaxton** [comment]
With the pro version the notification count still shows even when it's 0.
image
image

**Comment by shervElmi** [comment]
> With the pro version the notification count still shows even when it's 0. image image


Thank you for your review, @garretlaxton. I haven't been able to reproduce this issue. Could you kindly provide a QA URL where this issue is apparent?

**Comment by garretlaxton** [comment]
@shervElmi Sure!

https://bd.grunge.us.eu.org/wp-admin/
User: Sherv
Pass: vb3AwO#v5j#RNJN6fFBPU#I!

**Comment by shervElmi** [comment]
> @shervElmi Sure!
> 
> https://bd.grunge.us.eu.org/wp-admin/ User: Sherv Pass: vb3AwO#v5j#RNJN6fFBPU#I!

Thanks a lot, @garretlaxton! I've fixed the issue you mentioned, and now the notifications box shows up when the page first loads.

**Note:** Please update your QA BD plugin.

**Comment by garretlaxton** [comment]
Thanks! I see the notifications open now though.
image
image

**Comment by shervElmi** [comment]
> Thanks! I see the notifications open now though. image image

Thanks! I've checked your QA site, and it seems that the latest updates haven't been applied. I tried clearing the cache, but some code still doesn't appear to be updated.

**Comment by garretlaxton** [comment]
I primarily use my localhost for testing, I just use that site for sharing what I see. I have since updated the plugin there if you want to take a look now.

**Comment by shervElmi** [comment]
Thanks, @garretlaxton for your review. I've just updated the code to stop showing the `Hide Notifications` button when there's nothing to display.

**Review by garretlaxton — APPROVED** [review]
Nice, this looks good now!

File: includes/class-assets.php
Comment (mostafa-hisham) [inline includes/class-assets.php:501]: Thanks for your work @shervElmi 

I don't think we need to make this function static

**Review by mostafa-hisham — CHANGES_REQUESTED** [review]

File: includes/class-assets.php
Comment (shervElmi) [reply includes/class-assets.php:501]: Thank you for your review, @mostafa-hisham!

We apply Floating Links in all S11 plugins. I made this method static to avoid using any `$this` reference in the future. I believe the static approach works well here.

File: assets/css/packages/s11-floating-links.css
Comment (stephywells) [inline assets/css/packages/s11-floating-links.css:26]: The X and Percie aren't quite centered vertically in the circle. This css fixes it for me:
`.s11-svg-wrapper{display:flex}`

File: assets/css/packages/s11-floating-links.css
Comment (stephywells) [inline assets/css/packages/s11-floating-links.css:12]: Can we override this position in BD? I think it's 48px in FF because of the screen options tab. But here, it would looks better if it aligned with the notifications. It matches for me with 20px for right and bottom.

Before:
Screenshot 2023-08-04 at 6 41 23 PM

After:
Screenshot 2023-08-04 at 6 41 11 PM

**Review by stephywells — CHANGES_REQUESTED** [review]
This is working great! Nice job @shervElmi !

I'm seeing a few small styling issues, but it's super close.

**Comment by shervElmi** [comment]
Thank you for your review @stephywells. I hadn't realized you were on this PR; my apologies for that oversight!

File: assets/css/packages/s11-floating-links.css
Comment (stephywells) [reply assets/css/packages/s11-floating-links.css]: This is almost there. It's just a bit too high now. Here are the current links with the right spacing:
Screenshot 2023-08-29 at 5 53 05 PM

But this is spaced too high. It should be the same 16px as the above screenshot:
Screenshot 2023-08-29 at 5 53 52 PM

**Comment by shervElmi** [comment]
Thank you for your input on this, @stephywells and @mostafa-hisham. I've addressed the issues you mentioned. Please let me know if you need any further changes.

**Review by stephywells — CHANGES_REQUESTED** [review]
Thanks @shervElmi this is almost there now. I'm just checking the last details.
Screenshot 2023-08-30 at 2 01 15 PM

Can we update the colors to match BD too?
--floating-links-bg-hover-color: rgba(109,135,185,0.07);
--floating-links-hover-color: #3C4B5D;
.s11-floating-links-nav-text{color: rgba(60,75,93,0.8);}

Can we also set the X to rgba(60,75,93,0.8)? We could set fill="currentColor" in the svg so we can change it in the css.

Percie feels a little too small to me. This is a mistake in the design with extra white space in the svg. Can we change the svg from:
width="31" height="40"
to:
width="35" height="46"

Also, I'm not seeing any utm on the links. Can we add those so we know if this is helping or not? 
- In FF, the links to our site should run through FrmAppHelper::admin_upgrade_link().
- In BD, it should go through wpbdp_admin_upgrade_link().

With these fixes, it should be completely ready to go. Thanks for fixing these @shervElmi !

File: includes/admin/helpers/class-notices.php
Comment (stephywells) [inline includes/admin/helpers/class-notices.php:34]: I'm seeing some leftover references to ".wpbdp-bell-notification" and "wpbdp-bell-notification-icon". Can we remove more of this?

**Review by stephywells — CHANGES_REQUESTED** [review]
Well... I was wrong. I have a few more requests. I hadn't reviewed the code in detail. Sorry about that!

File: includes/class-assets.php
Comment (shervElmi) [reply includes/class-assets.php:512]: Of course, I'll incorporate your input. My goal is to make updates easier to manage across multiple plugins. Since we don't have notifications in FF, I've decided to create a separate file that we can easily copy and paste if needed. The idea is to only change the `config.js` file and keep `s11-floating-links.js` and `s11-floating-links-notifications.js` the same across all plugins, maintaining consistent functionality.

Comment (shervElmi) [reply includes/class-assets.php:512]: We can ignore this for now because I've researched how we could combine them. Implementing this would require some setup in our plugin, which could be time-consuming. Below are the structures for your consideration. Please let me know what your thoughts are. Do you think we should proceed with this?

1. **Run the following commands to install `grunt-babel`, `@babel/core`, and `@babel/preset-env`:**

    ```bash
    npm install grunt-babel @babel/core @babel/preset-env --save-dev
    ```

2. **Update your `Gruntfile.js` to include a new Babel task. You'll add the following section:**

    ```javascript
    // Add these lines at the top of your Gruntfile.js
    const babel = require('grunt-babel');
    
    // ...existing code
    
    module.exports = function( grunt ) {
        // ...existing config
    
        // Add this inside the 'grunt.initConfig' function
        grunt.config.set('babel', {
            options: {
                sourceMap: true,
                presets: ['@babel/preset-env']
            },
            dist: {
                files: [
                    {
                        expand: true,
                        cwd: 'src/js/',  // replace with your source JavaScript files path
                        src: ['*.js'],
                        dest: 'dist/js/',  // replace with your transpiled files destination folder
                        ext: '.js'
                    }
                ]
            }
        });
    
        // ...existing tasks
    
        // Load grunt-babel
        grunt.loadNpmTasks('grunt-babel');
    
        // ...existing code
    };
    ```

3. **Add `babel` to your task lists wherever you'd like it to run. For example:**

    ```javascript
    // Inside your 'minify' task or any other task where you want to run Babel
    if ('undefined' !== typeof grunt.config.get('babel')) {
        grunt.task.run('babel');
    }
    ```

File: includes/class-assets.php
Comment (shervElmi) [inline includes/class-assets.php:521]: I've pushed a commit addressing your feedback @stephywells and covering all items, except for a few where I've added comments.
I have another question: When a user activates the premium version, they will be sent a link containing the `utm_campaign=liteplugin` parameter for tracking. Does this look good to you, or should it be changed?

File: includes/class-assets.php
Comment (stephywells) [reply includes/class-assets.php:508]: @shervElmi I don't think this would take much time, but I don't think we need to make it a priority right now. We can skip it since this is only on BD pages in the admin.

File: includes/admin/helpers/class-notices.php
Comment (stephywells) [reply includes/admin/helpers/class-notices.php:34]: I'm fine with that. Thank you!

Comment (stephywells) [reply includes/admin/helpers/class-notices.php:34]: @shervElmi I'm seeing css that isn't being used that would be nice to remove. It's much easier to do now than later.

File: includes/class-assets.php
Comment (stephywells) [reply includes/class-assets.php:521]: That's fine, since it's still in the lite plugin code. Thanks for checking!

File: assets/js/packages/floating-links/config.js
Comment (stephywells) [inline assets/js/packages/floating-links/config.js:112]: Looks like we can cut down the svg size quite a bit with https://jakearchibald.github.io/svgomg/

**Review by stephywells — APPROVED** [review]
This matches great, works great, and looks great!

Very nice work on this @shervElmi, and thinking through all the complexities of keeping it easy to maintain for all the products. Thanks so much!

I think we can merge this once my last comment is addressed on the SVG compression.

**Comment by shervElmi** [comment]
@stephywells, I appreciate your positive words. Thank you!
I just used the mentioned site to reduce the logo icon size.

**Comment by stephywells** [comment]
Thank you @shervElmi ! It looks great!

**Comment by stephywells** [comment]
@mostafa-hisham This one is just waiting on your review now.

**Comment by stephywells** [comment]
Yay! Go team!

**Pull request merged**
